### PR TITLE
Added IPSAR integrations

### PR DESCRIPTION
**Pull request by EmilNik**

### CODE REVIEW

File: drone_sar/ipsar_dataset.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    """
+    A PyTorch dataset for the IPSAR dataset.
+
+    Args:
+        root_dir (str): The path to the root directory of the IPSAR dataset.
+        set_name (str): The name of the dataset split ("train" or "test").
+    """
+
+    def __init__(self, root_dir: str, set_name: str):
+        assert set_name in ["train", "test"], f"Invalid set name: {set_name}"
```
Comment (ichko) [inline drone_sar/ipsar_dataset.py]: A better name would be `mode` as in the mode of the dataset - train or test. `set_name` sound like a verb at first glance :smile:. `split_type` is also good and more verbose.

File: drone_sar/ipsar_dataset.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                            (
+                                np.asarray(Image.open(img_path)),
+                                {
+                                    "boxes": annotation["boxes"][0],
+                                    "class_labels": annotation["class_labels"][0],
+                                },
+                            )
+                        )
+                    else:
+                        # TODO: Add empty dataset entry
```
Comment (ichko) [inline drone_sar/ipsar_dataset.py:50]: We might not need explicit negative examples as stated here: https://ai.stackexchange.com/questions/28371/how-to-add-negative-samples-for-object-detection

> The paper basically proposes to think that each pixel of a dataset image is a training signal. Then for each image there are lots of pixels with negative signal (nothing in it: sky, ground, trees...) and only a few with positive signal (the actual car).

We can later verify if this is the case by predicting on the empty images and asserting there are not confidently wrong predictions.

**Review by ichko — APPROVED** [review]
The changes look good to me. I am impressed by how well you have documented and formatted your code. You have also not missed some subtleties that I was looking for (like setting the correct model mode when training/evaluating). 

One thing that might be helpful is adding loss/metrics curve logging - we can utilize W&B (free for individuals) or mlflow (free for all). This will give us insight into how the quality of training is going and when would we like to before overfitting. 

Overall - great work, approved! :muscle:

File: drone_sar/ipsar_dataset.py (hunk 1 above)
Comment (EmilNik) [reply drone_sar/ipsar_dataset.py]: good point. it made sense at first, but now, looking at the code a couple of days later it isn't that clear anymore :D
